js/plugins/google-genai/src/googleai/index.ts

```diff
@@ -186,47 +107,66 @@ export function googleAIPlugin(options?: PluginOptions): GenkitPlugin {
   );
 }
 
+// All the known model names for intellisense completion
+type KnownModels = gemini.KnownModels | imagen.KnownModels;
+type KnownEmbedders = embedder.KnownModels;
+
 export type GoogleAIPlugin = {
-  (params?: PluginOptions): GenkitPlugin;
-  model(
-    name: GeminiVersionString,
-    config?: z.infer<typeof GeminiConfigSchema>
-  ): ModelReference<typeof GeminiConfigSchema>;
-  embedder(
-    name: string,
-    config?: GeminiEmbeddingConfig
-  ): EmbedderReference<typeof GoogleAIEmbeddingConfigSchema>;
+  (params?: GoogleAIPluginOptions): GenkitPlugin;
+  model<T extends string>(
+    name: T | KnownModels,
```

Comment:
did you try autocompletion for this? The known working way is like this:
https://github.com/firebase/genkit/blob/9ea22ac9ca8ffd5a42ba96d4390f41d70fdc9380/js/plugins/googleai/src/index.ts#L344

Reply:
Discussed offline